src/partitioning/parmetis_partitioner.C

```diff
@@ -185,13 +185,13 @@ void ParmetisPartitioner::_do_repartition (MeshBase & mesh,
                                        &itr,
                                        &_pmetis->options[0],
                                        &_pmetis->edgecut,
-                                       _pmetis->part.empty()    ? nullptr : &_pmetis->part[0],
+                                       _pmetis->part.empty()    ? nullptr : reinterpret_cast<Parmetis::idx_t *>(&_pmetis->part[0]),
```

Comment:
I suppose this avoids the warning because there's no dereferencing happening? Sounds like a reasonable fix to me, but I would mind if you also documented why `part` does not use `Parmetis::idx_t` like all the other things we're passing to Parmetis.

Reply:
The comment in the header isn't good enough?

Reply:
Ah, no, because that discusses passing it to parmetis not to libMesh.  Will fix.